kotlinx-coroutines-core/common/src/EventLoop.common.kt

```diff
@@ -64,11 +67,17 @@ internal abstract class EventLoop : CoroutineDispatcher() {
         }
 
     public fun processUnconfinedEvent(): Boolean {
-        val queue = unconfinedQueue ?: return false
-        val task = queue.removeFirstOrNull() ?: return false
+        val task = popUnconfinedTask() ?: return false
         task.run()
         return true
     }
+
+    public fun popUnconfinedTask(): DispatchedTask<*>? =
+        unconfinedQueue?.removeFirstOrNull()
+
+    protected open fun popNextTask(onlyUnconfined: Boolean = false): Runnable? =
+        popUnconfinedTask()
+
```

Comment:
Naming nit: `pop` verb is primarily used with stacks (`push`/`pop`), rarely so with queues, especially in JVM & Kotlin ecosystem. I'd suggest using `poll` (ala Java's Queue interface) for a short verb these operations or you can go with long and explicit `removeFirstXxxOrNull`.